### ISSUE TITLE
"docker stack deploy" does NOT care if additional volume keys are used alongside "external"

### ISSUE THREAD

**Issue by briantd** [docker/docker.github.io#3142]
File: [compose/compose-file/index.md](https://docs.docker.com/compose/compose-file/), CC @londoncalling

This issue concerns how `docker stack deploy` handles additional properties used alongside "external".
https://docs.docker.com/compose/compose-file/#external

Documentation says that "external" property must be the only one:
<img width="667" alt="screenshot of safari 5-2-17 4-40-25 pm" src="https://cloud.githubusercontent.com/assets/94055/25643815/a4f24d02-2f56-11e7-9fad-818620d40717.png">

However, "docker stack deploy" does NOT seem to care.

```
cat dc-test-external-plus-other.yml
version: "3.1"
services:
  aservice:
    image: busybox
    volumes:
      - test-vol:/tmp:z
volumes:
  test-vol:
    external: true
    driver: local

$ docker stack deploy -c dc-test-external-plus-other.yml myapp2
Creating network myapp2_default
Creating service myapp2_aservice
```

**Comment by londoncalling** [comment]
@shin- @dnephin @thaJeztah  is this limitation ^^ on `external` not relevant for `docker stack deploy` by design?

**Comment by dnephin** [comment]
I believe this was fixed here by #31176 which is in 17.04

**Comment by londoncalling** [comment]
Closing this issue per @dnephin and the referenced PR ^^
